flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -107,6 +108,8 @@
 	@Nullable
 	protected final BufferCompressor bufferCompressor;
 
+	private final ChannelStateReader channelStateReader;
+
```

Comment:
The same as for `PipelinedSubpartition`: I think it's better to pass `channelStateReader` as a parameter to `initializeState` (to reduce coupling and other benefits).

Reply:
Replied above.